### PR TITLE
Add unit test for duplicate ID in schema

### PR DESCRIPTION
**Pull request by nbdevos25**
Add unit test for duplicate ID in schema

- No code change
- Added missing unit test for duplicate ID in schema 
- Java already handles https://github.com/apache/iceberg-go/issues/593. I will publish a PR to fix the issue in iceberg-go repo

### CODE REVIEW

File: api/src/test/java/org/apache/iceberg/TestSchema.java [hunk 1]
```diff
@@ -120,6 +138,13 @@ public void testUnsupportedTypes(Type type, int unsupportedVersion) {
             MIN_FORMAT_VERSIONS.get(type.typeId()));
   }
 
+  @Test
+  public void testDuplicateIDs() {
+    assertThatThrownBy(() -> Schema.checkCompatibility(generateTypeSchemaDuplicateID(), 2))
```
Comment (ebyhr) [inline api/src/test/java/org/apache/iceberg/TestSchema.java:143]: The exception happens in `generateTypeSchemaDuplicateID`, not in `checkCompatibility`, right? This test seems a bit misleading since the outer method is redundant. 

Also, we already have `TestNameMapping#testFailsDuplicateId` to test duplicate ID from `Schema` constructor.

Comment (nbdevos25) [reply api/src/test/java/org/apache/iceberg/TestSchema.java:143]: thx. closing the PR
